src/ai/backend/storage/weka/weka_client.py

```diff
@@ -0,0 +1,388 @@
+            try:
+                if method == "GET" or method == "DELETE":
+                    return await func("/api/v2" + path, headers=self._req_header)
+
+                else:
+                    return await func(
+                        "/api/v2" + path,
+                        headers=self._req_header,
+                        data=body,
+                    )
+
+            except web.HTTPUnauthorized:
+                raise WekaUnauthorizedError
+
+    @error_handler
+    async def list_fs(self) -> Iterable[WekaFs]:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(sess, "GET", "/fileSystems")
+            data = await response.json()
+        if len(data["data"]) == 0:
+            raise WekaNotFoundError
+        return [WekaFs.from_json(fs_json) for fs_json in data["data"]]
+
+    @error_handler
+    async def get_fs(self, fs_uid: str) -> WekaFs:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(sess, "GET", f"/fileSystems/{fs_uid}")
+            data = await response.json()
+        if data.get("data") is None:
+            raise WekaNotFoundError
+        return WekaFs.from_json(data["data"])
+
+    @error_handler
+    async def list_quotas(self, fs_uid: str) -> Iterable[WekaQuota]:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "GET",
+                f"/fileSystems/{fs_uid}/quota",
+            )
+            data = await response.json()
+        if isinstance(data['data'], list):
+            return [
+                WekaQuota.from_json(quota_info["quota_id"], quota_info)
+                for quota_info in data["data"]
+            ]
+        else:
+            return [
+                WekaQuota.from_json(quota_id, data["data"][quota_id])
+                for quota_id in data["data"].keys()
+            ]
+
+    @error_handler
+    async def get_quota(self, fs_uid: str, inode_id: int) -> WekaQuota:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "GET",
+                f"/fileSystems/{fs_uid}/quota/{inode_id}",
+            )
+            data = await response.json()
+        if len(data["data"].keys()) == 0:
+            raise WekaNotFoundError
+        quota_id = data["data"].keys()[0]
+        return WekaQuota.from_json(quota_id, data["data"][quota_id])
+
+    @error_handler
+    async def set_quota(
+        self,
+        fs_uid: str,
+        inode_id: int,
+        soft_limit: BinarySize,
+        hard_limit: BinarySize,
+    ) -> WekaQuota:
+        def _format_size(s: BinarySize) -> str:
```

Comment:
We shall use `finite_from_str` function defined in `BinarySize` class instead of using inner function `_format_size`, since both `soft_limit_bytes` and `hard_limit_bytes` requires numbers in byte unit, not human-readable value, such as 2.5 GiB(2684354560 bytes).

Reply:
You're right. We don't need to change it. let's just leave them as is.